File: KeyboardTextFieldDemo/KeyboardTextFieldDemo/TextViewSpecialCaseViewController.m
//
//  TextViewSpecialCaseViewController.m
//  KeyboardTextFieldDemo

#import "TextViewSpecialCaseViewController.h"
#import "IQKeyboardManager.h"

@interface TextViewSpecialCaseViewController ()

-(void)refreshUI;

@end

@implementation TextViewSpecialCaseViewController

-(IBAction)canAdjustTextView:(UIBarButtonItem*)barButton
{
    if ([[IQKeyboardManager sharedManager] canAdjustTextView])
    {
        [[IQKeyboardManager sharedManager] setCanAdjustTextView:NO];
    }
    else
    {
        [[IQKeyboardManager sharedManager] setCanAdjustTextView:YES];
    }
    
    [self refreshUI];
}

#pragma mark - View lifecycle

- (void)viewDidLoad
{
    [super viewDidLoad];

    if (!self.navigationController)
    {
        [buttonPush setHidden:YES];
        [buttonPresent setTitle:@"Dismiss" forState:UIControlStateNormal];
    }
}

-(void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    [self refreshUI];
}

-(void)refreshUI
{
    if ([[IQKeyboardManager sharedManager] canAdjustTextView])
    {
        [barButtonAdjust setTitle:@"Disable Adjust"];
    }
    else
    {
        [barButtonAdjust setTitle:@"Enable Adjust"];
    }
}

-(BOOL)textView:(UITextView *)textView shouldChangeTextInRange:(NSRange)range replacementText:(NSString *)text
{
    if([text isEqualToString:@"\n"])
        [textView resignFirstResponder];
    return YES;
}

- (IBAction)presentClicked:(id)sender
{
    @try {
        if (self.navigationController)
        {
            TextViewSpecialCaseViewController *controller = [[TextViewSpecialCaseViewController alloc] init];
            
            [controller setModalTransitionStyle:arc4random()%4];
            
            // TransitionStylePartialCurl can only be presented by FullScreen style.
            if (controller.modalTransitionStyle == UIModalTransitionStylePartialCurl)
                controller.modalPresentationStyle = UIModalPresentationFullScreen;
            else
                controller.modalPresentationStyle = arc4random()%4;
            
            [self presentViewController:controller animated:YES completion:nil];
            
        }
        else
        {
            [self dismissViewControllerAnimated:YES completion:nil];
        }
    }
    @catch (NSException *exception) {
        NSLog(@"Exception:%@",exception);
    }
    @finally {
        
    }
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    return YES;
}

- (BOOL)shouldAutorotate
{
    return YES;
}

@end

File: KeyboardTextFieldDemo/IQKeyboardManager Swift/TextViewSpecialCaseViewController.swift
//
//  TextViewSpecialCaseViewController.swift
//  IQKeyboard
//
//  Created by Iftekhar on 23/09/14.
//  Copyright (c) 2014 Iftekhar. All rights reserved.
//

import Foundation
import UIKIt

class TextViewSpecialCaseViewController: UIViewController, UITextViewDelegate {
    
    @IBOutlet private var buttonPop : UIButton!;
    @IBOutlet private var buttonPush : UIButton!;
    @IBOutlet private var buttonPresent : UIButton!;
    @IBOutlet private var barButtonAdjust : UIBarButtonItem!;
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        if IQKeyboardManager.sharedManager().canAdjustTextView {
            barButtonAdjust.title = "Disable Adjust"
        } else {
            barButtonAdjust.title = "Enable Adjust"
        }
        
        if navigationController == nil {
            buttonPop.hidden = true
            buttonPush.hidden = true
            buttonPresent.setTitle("Dismiss", forState: UIControlState.Normal)
        }
    }
    
    
    override func viewWillAppear (animated : Bool) {
        
        super.viewWillAppear(animated)
        IQKeyboardManager.sharedManager().shouldToolbarUsesTextFieldTintColor = true
    }
    
    override func viewWillDisappear (animated : Bool) {
        
        super.viewWillDisappear(animated)
        IQKeyboardManager.sharedManager().shouldToolbarUsesTextFieldTintColor = true
    }
    
    @IBAction func canAdjustTextView (barButton : UIBarButtonItem!) {
        
        if barButton.title == "Disable Adjust" {
            IQKeyboardManager.sharedManager().canAdjustTextView = false
            barButton.title = "Enable Adjust"
        } else {
            IQKeyboardManager.sharedManager().canAdjustTextView = true
            barButton.title = "Disable Adjust"
        }
    }
    
    func textView(textView: UITextView, shouldChangeTextInRange range: NSRange, replacementText text: String) -> Bool {
        
        if text == "\n" {
            textView.resignFirstResponder()
        }
        
        return true
    }
    
    @IBAction func popClicked (barButton : UIButton!) {
        navigationController?.popViewControllerAnimated(true)
    }
    
    @IBAction func presentClicked (barButton : UIButton!) {
        
        if (navigationController) != nil {
            var controller : TextViewSpecialCaseViewController = TextViewSpecialCaseViewController()
            presentViewController(controller, animated: true, completion: nil)
        } else {
            dismissViewControllerAnimated(true, completion: nil)
        }
    }
}

